### ISSUE TITLE
Local failing functional test: communication preferences - opt-in on signup

### ISSUE THREAD

**Issue by shane-tomlinson** [mozilla/fxa-content-server#5433]
> × firefox on any platform - communication preferences - opt-in on signup (2.955s)
Error: [POST http://localhost:4444/wd/hub/session/899e2b9b-efd5-437f-b2af-fbe4e6968d09/element/20/click / {}]
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'stomlinson-30887', ip: '192.168.0.22', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.6', java.version: '1.8.0_131'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Capabilities [{moz:profile=/var/folders/cq/drrh74v57_x_x6ksf_7pk2x80000gn/T/rust_mozprofile.nvlJ14CAYKkC, rotatable=false, timeouts={implicit=0.0, pageLoad=300000.0, script=30000.0}, pageLoadStrategy=normal, platform=ANY, specificationLevel=0.0, moz:accessibilityChecks=false, acceptInsecureCerts=false, browserVersion=54.0.1, platformVersion=16.7.0, moz:processID=15969.0, browserName=firefox, javascriptEnabled=true, platformName=darwin}]
Session ID: 9989b813-af55-ff42-a0cb-a059566f4d44
  at Command.<anonymous>  <tests/functional/lib/helpers.js:184:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:56:25>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:1300:64>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:1298:8>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:56:25>
  at Test.opt-in on signup [as test]  <tests/functional/email_opt_in.js:80:10>

Selenium cannot interact with the checkbox since it's hidden. Ref #5305.

**Comment by shane-tomlinson** [comment]
The odd thing is, if I update the test to click on the span, the test also fails because the user isn't signed in. Even more concerning, if I go through the flow manually in Firefox 55, users are not opted into marketing email because Fx 55 does not believe the checkbox is clicked.

Putting a heart on this. Labeling train-95 until we can figure out if this is a real problem.

**Comment by shane-tomlinson** [comment]
I just tested with Firefox 55 against prod. The opt-in is being captured and sent along with the call to /recovery_email/verify_code. That's a good start. Doesn't explain this local failing test though.

**Comment by shane-tomlinson** [comment]
Because this is being captured in prod, this is lower priority than I initially feared. Bumping from train-95 to train-96.

**Comment by shane-tomlinson** [comment]
Turns out the issue is when developing locally, that the fake basket server never receives the subscribe notification because we aren't running a message queue.

**Comment by shane-tomlinson** [comment]
There are a couple of content-server problems too, I'll file a PR with the changes needed.

1. Click on the opt-in checkbox does not check the checkbox.
2. waitForBasket is [wrapped in a thenify](https://github.com/mozilla/fxa-content-server/blob/57568aa13f41ad4b9d0b871e4ae95567a9948774/tests/functional/email_opt_in.js#L34) and shouldn't be. It already returns a function which returns a promise.

**Comment by shane-tomlinson** [comment]
Ref https://github.com/mozilla/fxa-auth-server/issues/2112

**Comment by shane-tomlinson** [comment]
Blocked on an auth-server/basket-server comm mechanism.

**Comment by vladikoff** [comment]
Could be unblocked by https://github.com/mozilla/fxa-local-dev/issues/94

**Comment by shane-tomlinson** [comment]
> With #5551, this should no longer be a problem.

I was wrong, with #5551, this is still a problem. No SQS.

**Comment by vladikoff** [comment]
Still no SQS in local dev!
